src/actions/authActions.js

```diff
@@ -63,23 +67,39 @@ export function autologin(jwt, doGoBack) {
   };
 }
 
-export function login(username, password) {
+export function login(token_id) {
   return (dispatch) => {
     loginMutate({
       variables: {
-        user: {
-          username,
-          password
-        }
+        token_id
       }
     }).then((res) => {
-      const jwt = res.data.loginUser.payload;
-      if (jwt) {
-        dispatch(autologin(jwt, true));
-      } else {
-        dispatch({
-          type: LOGIN_FAILED
-        });
+      console.log(res);
+      const status = res.data.loginUser.status;
+      switch (status) {
+        case LoginResults.SUCCESS:
+          const jwt = res.data.loginUser.jwt;
+          dispatch(autologin(jwt, true));
+          break;
+        case LoginResults.NEED_APPROVAL:
+          dispatch({
```

Comment:
generally it is recommended to use action creators. and as mentioned above - those "error" cases can be merged.